File: backend/src/main/java/com/bookflow/catalog/service/CatalogService.java
package com.bookflow.catalog.service;

import com.bookflow.catalog.dto.request.CreateCatalogRequest;
import com.bookflow.catalog.dto.request.UpdateCatalogRequest;
import com.bookflow.catalog.dto.response.CatalogResponse;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.List;

public interface CatalogService {

    CatalogResponse create(Long companyId, CreateCatalogRequest request);
    CatalogResponse findById(Long companyId, Long id);
    List<CatalogResponse> findAllByCompany(Long companyId);
    Page<CatalogResponse> findAllByCompanyPaged(Long companyId, Pageable pageable);
    List<CatalogResponse> findAll();
    List<CatalogResponse> findAllIncludingInactive();
    CatalogResponse update(Long companyId, Long id, UpdateCatalogRequest request);
    void delete(Long companyId, Long id);
    void activate(Long companyId, Long id);
}

File: frontend/src/services/catalogService.js
import api from './api'

export const catalogService = {
  create: (companyId, data) =>
    api.post(`/api/v1/companies/${companyId}/catalog`, data),

  update: (companyId, id, data) =>
    api.put(`/api/v1/companies/${companyId}/catalog/${id}`, data),

  getById: (companyId, id) =>
    api.get(`/api/v1/companies/${companyId}/catalog/${id}`),

  getAll: (companyId) =>
    api.get(`/api/v1/companies/${companyId}/catalog`),

  getPaged: (companyId, page = 0, size = 20) =>
    api.get(`/api/v1/companies/${companyId}/catalog/paged`, {
      params: { page, size },
    }),

  delete: (companyId, id) =>
    api.delete(`/api/v1/companies/${companyId}/catalog/${id}`),

  activate: (companyId, id) =>
    api.patch(`/api/v1/companies/${companyId}/catalog/${id}/activate`),
}
